joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -822,14 +827,21 @@ controller_interface::CallbackReturn JointTrajectoryController::on_activate(
   last_state_publish_time_ = get_node()->now();
 
   // Initialize current state storage if hardware state has tracking offset
-  resize_joint_trajectory_point(last_commanded_state_, joint_names_.size());
+  resize_joint_trajectory_point(state_current_, dof_);
+  resize_joint_trajectory_point(state_desired_, dof_);
+  resize_joint_trajectory_point(state_error_, dof_);
+  resize_joint_trajectory_point(last_commanded_state_, dof_);
```

Comment:
can we resize those in `on_configure`? Since this should be, if possible, real-time capable method.

Reply:
That makes sense. Done and tested.